app/services/data_set_documentation.rb

```diff
@@ -2,35 +2,27 @@ class DataSetDocumentation
   def self.for(klass)
     name = klass.name.demodulize.underscore
 
-    begin
-      spec = YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
-    rescue Errno::ENOENT
-      return false
-    end
+    spec = YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
+    common_columns = load_common_columns
+    custom_columns = spec['custom_columns']
 
-    common_columns = Dir[Rails.root.join('app/exports/common_columns/*')]
-                         .map { |file| YAML.load_file(file) }
-                         .reduce({}, :merge)
+    check_for_shadowed_columns(common_columns, custom_columns)
 
     used_common_columns = common_columns.slice(*spec['common_columns'])
-    used_common_columns.merge(spec['custom_columns'] || {})
+    used_common_columns.merge(custom_columns || {})
   end
 
-  def self.shadowed_common_columns(klass)
-    name = klass.name.demodulize.underscore
-
-    begin
-      spec = YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
-    rescue Errno::ENOENT
-      return false
+  def self.check_for_shadowed_columns(common_columns, custom_columns)
+    if (custom_columns.keys.map(&:to_sym) & common_columns.keys.map(&:to_sym)).any?
+      raise 'There are columns in the export documentation that shadow the shared common columns'
```

Comment:
I wasn't sure if this was the correct way to raise exceptions. But I've seen similar examples in the codebase.